tests/lib/external/snapd-testing-tools/utils/log-parser

```diff
@@ -160,17 +126,21 @@ class Rule:
     """
 
     def __init__(self, rule):
-        parts = rule.split('=',1)
+        parts = rule.split("=", 1)
         if len(parts) != 2:
-            raise ValueError("Error: Rule '{}' does not follow the KEY=PATTERN format".format(rule))
+            raise ValueError(
+                "Error: Rule '{}' does not follow the KEY=PATTERN format".format(rule)
+            )
 
         self.key = parts[0]
         self.pattern = parts[1]
 
         try:
             re.compile(self.pattern)
```

Comment:
@zyga wrote:
You are throwing away the result of the compilation. Just store it as an instance attribute and remove all the pointless re-compilation form the filter method.